File: Bubble_Sort/PHP/jakobhans/bubble_sort_test.php
<?php
require "bubble_sort.php"; 

class BubbleSortTest extends PHPUnit_Framework_TestCase{
    // Given an unorder array
    private $test = [4,6,1,6,5,9,4,100,-4,50];

    public function testOrderAnArray(){
        $sorted = $this->test; 
        sort($sorted, SORT_NUMERIC);
        $this->assertEquals($sorted, bubble_sort($this->test));
    }

    public function testNoParameters(){
        $this->assertEmpty(bubble_sort([]));
    }
}

File: Bubble_Sort/Scala/aayushKumarJarvis/BubbleSortTest.scala
import org.scalatest.{Matchers, FunSuite}

class BubbleSortTest extends FunSuite with Matchers {

  test("Bubble Sort should sort") {
    val objectForBubbleSort = new BubbleSort

    objectForBubbleSort.bubbleSort(Array(11,1,4,22)) should be(Array(1,4,11,22))
    objectForBubbleSort.bubbleSort(Array(1,2,3,4,5)) should be(Array(1,2,3,4,5))
  }
}

